masq/src/commands/financials_command.rs

```diff
@@ -725,64 +564,120 @@ impl FinancialsCommand {
             },
         ]
         .into_iter()
-        .map(assemble_optional_strings_into_single_number)
+        .map(assemble_segments_into_single_number)
         .collect::<VecDeque<String>>();
-        compose_ranges(vec_of_correct_values)
+        compose_ranges(vec_of_possibly_corrected_values)
     }
 
-    fn no_records_found<F>(stdout: &mut dyn Write, headings: &[&str], write_headings: F)
-    where
-        F: Fn(&mut dyn Write, &[(&str, &usize)]),
-    {
-        let headings_widths = Self::widths_of_str_values(headings);
-        write_headings(stdout, &Self::zip_them(headings, &headings_widths));
+    fn triple_or_single_blank_line(stdout: &mut dyn Write, leading_dump: bool) {
+        if leading_dump {
+            short_writeln!(stdout)
+        } else {
+            short_writeln!(stdout, "\n\n")
+        }
+    }
+
+    fn no_records_found(stdout: &mut dyn Write, headings: &[String]) {
+        let mut headings_widths = Self::widths_of_str_values(headings);
+        let _ = mem::replace(&mut headings_widths[1], WALLET_ADDRESS_LENGTH);
+        Self::write_column_formatted(stdout, &Self::zip_them(headings, &headings_widths));
         short_writeln!(stdout, "\nNo records found",)
     }
 
+    fn width_precise_calculation(
+        headings: &HeadingsHolder,
+        values_of_accounts: &[Vec<String>],
+    ) -> Vec<usize> {
+        let headings_widths = Self::widths_of_str_values(headings.words.as_slice());
+        let values_widths = Self::figure_out_max_widths(values_of_accounts);
+        Self::yield_bigger_values_from_vecs(headings_widths, &values_widths)
+    }
+
+    fn widths_of_str_values<T: AsRef<str>>(headings: &[T]) -> Vec<usize> {
+        headings
+            .iter()
+            .map(|phrase| phrase.as_ref().len())
+            .collect()
+    }
+
+    fn render_accounts_generic<A: StringValuesFormattableAccount>(
+        &self,
+        stdout: &mut dyn Write,
+        accounts: Vec<A>,
+        headings: &HeadingsHolder,
+    ) {
+        let preformatted_subset = &accounts
+            .iter()
+            .enumerate()
+            .map(|(idx, account)| account.string_values(idx + 1, headings.is_gwei))
+            .collect::<Vec<_>>();
+        let optimal_widths = Self::width_precise_calculation(headings, preformatted_subset);
+        let zipped_inputs = &Self::zip_them(headings.words.as_slice(), &optimal_widths);
```

Comment:
Maybe a better name for this variable? Maybe headings_and_widths or something similar?